dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEnrollmentAnalyticsService.java

```diff
@@ -157,19 +166,25 @@ protected Grid createGridWithHeaders(EventQueryParams params) {
   protected long addEventData(Grid grid, EventQueryParams params) {
     Timer timer = new Timer().start().disablePrint();
 
-    params = queryPlanner.planEnrollmentQuery(params);
+    List<EventQueryParams> paramsList;
 
-    timer.getSplitTime("Planned event query, got partitions: " + params.getPartitions());
+    if (params.getEndpointAction() == RequestTypeAware.EndpointAction.AGGREGATE) {
+      paramsList = queryPlanner.planAggregateQuery(params);
+    } else {
+      paramsList = List.of(queryPlanner.planEnrollmentQuery(params));
+    }
 
     long count = 0;
+    for (EventQueryParams queryParams : paramsList) {
+      timer.getSplitTime("Planned event query, got partitions: " + queryParams.getPartitions());
+      if (queryParams.isTotalPages()) {
```

Comment:
I'm not sure if we should be counting and handling pagination as this is aggregated data only. But this is part of our discussion regarding the renaming of the method, let's see...

Reply:
done